types/leaflet/index.d.ts

```diff
@@ -2085,14 +2085,18 @@ export interface CircleMarkerOptions extends PathOptions {
     radius: number;
 }
 
+export interface CircleMarkerStyleOptions extends PathOptions {
+    radius?: number | undefined;
+}
+
 export class CircleMarker<P = any> extends Path {
     constructor(latlng: LatLngExpression, options: CircleMarkerOptions);
     toGeoJSON(precision?: number | false): geojson.Feature<geojson.Point, P>;
     setLatLng(latLng: LatLngExpression): this;
     getLatLng(): LatLng;
     setRadius(radius: number): this;
     getRadius(): number;
-    setStyle(options: CircleMarkerOptions): this;
+    setStyle(options: CircleMarkerStyleOptions): this;
```

Comment:
```suggestion
    setStyle(options: Partial<CircleMarkerOptions>): this;
```

See https://www.typescriptlang.org/docs/handbook/utility-types.html#partialtype